### PR TITLE
Verify client upgrade

### PR DESCRIPTION
**Pull request by zajck**
Main purpose of this PR are the tests that verfies that client upgrade was succesful (i.e. upgrade does not corrupt the data and new functions work).

Same test pattern is used as for protocol upgrade:
- generic tests the could be reused for future upgrades: `test/upgrade/clients/01_generic.js`
- this upgrade specific tests in `test/upgrade/clients/BosonVoucher-2.1.0-2.2.0.js`
 
Beside that some other changes were made:
- `getInterfaceIds` was changed so it works also with interfaces that inherit other interfaces (before we had to hardcode this)
- `IERC2981` was added so `getInterfaceIds` can calculate its id and is not hardcoded anymore
- `scripts/upgrade-facets.js` was changed so it uses uncached "./util/deploy-protocol-handler-facets.js"

### CODE REVIEW

**Review by anajuliabit — COMMENTED** [review]
Looks pretty good, just a few things

**Review by cliffhall — DISMISSED** [review]
LGTM! 👍

File: scripts/upgrade-facets.js [hunk 1]
```diff
@@ -392,4 +392,9 @@ const logFacetCut = (cut, selectors) => {
   }
 };
 
+function requireUncached(module) {
```
Comment (anajuliabit) [inline scripts/upgrade-facets.js:395]: Could you explain why this method is needed?

Comment (zajck) [reply scripts/upgrade-facets.js:413]: Added explanation to the code.

File: test/upgrade/00_config.js [hunk 2]
```diff
@@ -38,7 +38,24 @@ const facets = {
...
+        "ProtocolInitializationHandlerFacet",
+        "OfferHandlerFacet",
+        "ExchangeHandlerFacet",
+        "ConfigHandlerFacet",
+        "DisputeResolverHandlerFacet",
+      ],
+      remove: [],
+      skipSelectors: {},
+      facetsToInit: {},
+      initializationData: "0x0000000000000000000000000000000000000000000000000000000000005555",
```
[the remark this answers is not in the document]
Comment (zajck) [reply test/upgrade/00_config.js]: Comment added

[the remark this answers is not in the document]
Comment (zajck) [reply test/upgrade/00_config.js]: And btw, this config is not final yet. You'll have to add other facets to `addOrUpgrade` when you'll work on protocol upgrade tests.

**Review by anajuliabit — DISMISSED** [review]
LGTM

File: scripts/upgrade-facets.js (hunk 1 above)
Comment (anajuliabit) [reply scripts/upgrade-facets.js:413]: thank you

**Review by cliffhall — APPROVED** [review]
LGTM! 👍

**Pull request merged**